pkg/common/cns-lib/volume/manager.go

```diff
@@ -2626,19 +2690,38 @@ func (m *defaultManager) deleteSnapshotWithImprovedIdempotencyCheck(
 	}()
 
 	if deleteSnapshotTask == nil {
+
 		deleteSnapshotTask, err = invokeCNSDeleteSnapshot(ctx, m.virtualCenter, volumeID, snapshotID)
 		if err != nil {
 			if cnsvsphere.IsNotFoundError(err) {
 				log.Infof("SnapshotID: %q on volume with volumeID: %q, not found, thus returning success",
 					snapshotID, volumeID)
+				if isStorageQuotaM2FSSEnabled && m.clusterFlavor == cnstypes.CnsClusterFlavorWorkload {
+					//  return updated cnssnapshotinfo
+					aggregatedSnapshotCapacityInMb, err := m.getAggregatedSnapshotSize(ctx, volumeID)
+					if err != nil {
+						return nil, logger.LogNewErrorf(log,
+							"Failed to get aggregated snapshot size for volume %q with error %v", volumeID, err)
+					}
+					cnsSnapshotInfo = &CnsSnapshotInfo{
+						SnapshotID:                          snapshotID,
+						SourceVolumeID:                      volumeID,
+						AggregatedSnapshotCapacityInMb:      aggregatedSnapshotCapacityInMb,
+						SnapshotLatestOperationCompleteTime: time.Now(),
+					}
+					aggregatedSnapshotCapacity := resource.NewQuantity(aggregatedSnapshotCapacityInMb, resource.BinarySI)
+					quotaInfo.AggregatedSnapshotSize = aggregatedSnapshotCapacity
+					quotaInfo.SnapshotLatestOperationCompleteTime.Time = time.Now()
```

Comment:
`cnsSnapshotInfo` and `quotaInfo` will have different time since you invoke `time.Now()` separately.